manticore/platforms/evm.py

```diff
@@ -801,6 +836,22 @@ def _rollback(self):
         self._pc = last_pc
         self._checkpoint_data = None
 
+    def _set_check_jmpdest(self, flag=True):
+        self._check_jumpdest = flag
+
+    def _check_jmpdest(self):
+        if issymbolic(self._check_jumpdest):
+            should_check_jumpdest = solver.get_all_values(self.constraints, self._check_jumpdest)
+            if len(should_check_jumpdest) != 1:
+                raise EthereumError("Conditional not concretized at JMPDEST check")
+        else:
+            should_check_jumpdest = [self._check_jumpdest]
+
+        if should_check_jumpdest == [True]:
```

Comment:
Since we only support a symbolic case where it is concretized/has one solution (?) then maybe we should do retrieve the value in that branch:
```python
if len(should_check_jumpdest) != 1:
    raise EthereumError(...)

should_check_jumpdest = should_check_jumpdest[0]
```

and just assign `should_check_jumpdest = self._check_jumpdest` if it is not symbolic and then check it here as `if should_check_jumpdest is True:`